src/reflect/scala/reflect/internal/SymbolTable.scala

```diff
@@ -196,11 +197,13 @@ abstract class SymbolTable extends macros.Universe
   final def pushPhase(ph: Phase): Phase = {
     val current = phase
     phase = ph
-    phStack ::= ph
+    phStack(phStackIndex) = ph
```

Comment:
The alternative would be to use an `ArrayBuffer` and not worry about a hard limit to the size. I was being a bit paranoid with performance.